### ISSUE TITLE
Time travel

### ISSUE THREAD

**Issue by tibetiroka** [endless-sky/endless-sky#8936]
### Problem Description

I'm writing a PR that adds engine support for advancing the date by more than one day at a time, or even moving it backwards. This would give us the ability to, for example, slowly advance time during a long mission, without even leaving the conversation. (I actually plan using this feature in a mission.)

What needs most discussion is these points:
1. To what degree should moving the date backwards be supported?
2. How and where should this be represented in the data format?

### Related Issue Links

#8937 

### Desired Solution

**Currently, my PR uses this solution to the first problem:**

- Change the date back and do nothing. This is in contrast to moving forward in time, where all salaries and missions are handled properly.

**Reasons for choosing this route:**

While calculating salaries, tributes etc. should make it possible to extrapolate how much FL could have made in the past x days, there is no way to get it exactly right without actually storing this data for every single day. Also, keeping track of the account history and credit score would become pretty difficult; and restoring failed timed missions would be a nightmare.

This is why I think going about this as "restoring FL to their past version" is the wrong way to go about this. FL gets transported back to the past, more like.

Money can only be spent once. If FL gets transported back, they shouldn't get their money back, because that's already been spent. They get the opportunity to earn more money by living the same days, though.

On the other hand, if FL is moved forward in time, the banks would still want their due. Not being in the universe is a lame excuse for missing payments, and pug magic is not an acceptable currency. FL should feel lucky for not getting hit with any late fees.

**Ideas for the second problem:**
I see two major options here. We either:
- Create new keywords to set or offset the current date, or
- Make the date work like any other condition, supporting `=`, `+=` etc.

Personally, I think the second option is more versatile, but it might cause issues if there is already a `date` condition used by some plugin.

### Alternative Approaches

See above.

I'm not sure if we will ever need backwards time travel. I think it's nice for plugins, but it could be pretty easily removed from the PR, if necessary.

**Comment by TomGoodIdea** [comment]
related: #6018

**Comment by Hurleveur** [comment]
I dont think we should lean too much into time travel as we could easily get paradoxes and whatnot (like you know about the bombing so should be able to warn them and manyyyyy other things)
Just supporting going forward and giving possibility for backwards for plugins would be best I think.

**Comment by GPaso** [comment]
Best way of backwards time travel is plant a save and reload it.
No paradoxes, no complications. 

And for a way to go forward in time, instead of hopping in and out of a system or a planet, just add a "Sleep for today" button at the Player inventory.

**Comment by tigercoding56** [comment]
maybe it could be mission specific (so only certain missions can trigger time travel) so you could make a plugin allowing the user to warn of the bombing  , and deal with any complications yourself (for instance by having the time travel device be unstable and return you to current time after doing whatever you are supposed to be doing)

**Comment by ziproot** [comment]
Should this be closed as completed?

**Comment by petervdmeer** [comment]
> Should this be closed as completed?

No, only the engine support was added. So this ticket should still be open.
